Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -125,6 +125,80 @@ final class CacheAdvanceTests: XCTestCase {
         }
     }
 
+    private func prepareCacheWithIncorrectOffsetInHeader() throws -> CacheHeaderHandle {
+        let randomHighValue: UInt64 = 10_1000
+        let header = try CacheHeaderHandle(
+            forReadingFrom: testFileLocation,
+            maximumBytes: randomHighValue,
+            overwritesOldMessages: true)
+        let cacheToWrite = CacheAdvance<TestableMessage>(
+            fileURL: testFileLocation,
+            writer: try FileHandle(forWritingTo: testFileLocation),
+            reader: try CacheReader(forReadingFrom: testFileLocation),
+            header: try CacheHeaderHandle(
+                forReadingFrom: testFileLocation,
+                maximumBytes: header.maximumBytes,
+                overwritesOldMessages: header.overwritesOldMessages),
+            decoder: JSONDecoder(),
+            encoder: JSONEncoder())
+
+        let message = TestableMessage(stringLiteral: TestableMessage.lorumIpsum[0].value)
+        try cacheToWrite.append(message: message)
+
+        // Make sure the header data is persisted before we read it as part of the `messages()` call below.
+        try header.synchronizeHeaderData()
+        // Our file is empty. Make the file corrupted by setting the offset at end of newest message to be further in the file.
+        // This should never happen, but past versions of this repo could lead to a file having this kind of inconsistency if a crash occurred at the wrong time.
```

Comment:
If I'm understanding this properly, the infinite loop case above can _only_ be caused by file corruption due to a prior version of the library?

Reply:
Yes I'd love confirmation here too.